addons/account/models/account_analytic_account.py

```diff
@@ -100,3 +101,18 @@ def action_view_vendor_bill(self):
             'view_mode': 'tree,form',
         }
         return result
+
+    @api.ondelete(at_uninstall = False)
+    def _unlink_except_account_set_on_account_move_line(self):
+        query = self.env['account.move.line']._search([])
+        query.order = None
+        query.add_where(
+            'account_move_line.analytic_distribution ?| %s',
+            [[str(account_id) for account_id in self.ids]],
```

Comment:
this seemed a little dangerous, but this seems correctly escaped in the resulting query:

```sql
self = self.env['account.analytic.account'].browse(["h'ell[]b\\'obby table\"o"])
# results in
SELECT "account_move_line".id FROM "account_move_line" WHERE account_move_line.analytic_distribution ?| ARRAY['h''ell[]bobby table"o'] ORDER BY  "account_move_line"."date" DESC,"account_move_line"."move_name" DESC,"account_move_line"."id"
```